pkg/resources/envoy/envoy.go

```diff
@@ -15,23 +15,31 @@
 package envoy
 
 import (
+	"fmt"
+
 	"github.com/banzaicloud/kafka-operator/api/v1beta1"
 	"github.com/banzaicloud/kafka-operator/pkg/k8sutil"
 	"github.com/banzaicloud/kafka-operator/pkg/resources"
+	"github.com/banzaicloud/kafka-operator/pkg/util"
 	envoyutils "github.com/banzaicloud/kafka-operator/pkg/util/envoy"
-
 	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	componentName            = "envoy"
 	envoyVolumeAndConfigName = "envoy-config"
 	envoyDeploymentName      = "envoy"
+	envoyGlobal              = "envoy-global"
 )
 
-var labelSelector = map[string]string{
-	"app": "envoy",
+func labelSelector(envoyConfig *v1beta1.EnvoyConfig) map[string]string {
+	if envoyConfig.Id == envoyGlobal {
+		return map[string]string{"app": componentName}
+	} else {
+		return map[string]string{"app": fmt.Sprintf("%s-%s", componentName, envoyConfig.Id)}
```

Comment:
should we actually keep the same `app: envoy` for this too and additional add `group: envoy-{g1,g2..}` 
I am thinking that this would be useful to also query all envoy pods as well regardless of how many groups the user configures

Reply:
Agree. I also considered this approach.